ex5/FileSystemTPL/TPLOperations/MoveTPL.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+
+namespace FileSystemTPL.TPLOperations
+{
+    class MoveTPL : Operation
+    {
+        public MoveTPL(CommandDetails commandDetails, ExecutionManager executionManager) :
+            base(commandDetails, executionManager)
+        { }
+
+        public override void Execute()
+        {
+            string source = Path.Combine(Program.DIRECTORY_PATH, commandDetails.CommandArgs.First());
+            string destination = Path.Combine(Program.DIRECTORY_PATH, commandDetails.CommandArgs.Last());
+
+            Action<string, string> MovingFunction = GetMovingFunction(source);
+            MovingFunction(source, destination);
+        }
+
+        private Action<string, string> GetMovingFunction(string source)
+        {
+            try
+            {
+                executionManager.LockPath(source, ObjectInActionStatus.ToBeMoved);
+                
+                if (File.Exists(source))
+                {
+                    return MoveFileTPL;
+                }
+
+                else if (Directory.Exists(source))
+                {
+                    return MoveDirRecursively;
+                }
+
```

Comment:
usually blank lines does not show up between `if-else`, `try-catch-finally` because all of them are part of same unit code